crates/red_knot/src/symbols.rs

```diff
@@ -233,7 +266,12 @@ impl SymbolTable {
             .unwrap_or_default()
     }
 
-    fn add_symbol_to_scope(&mut self, scope_id: ScopeId, name: &str) -> SymbolId {
+    fn add_symbol_to_scope(
```

Comment:
I would say go ahead and rename this to `add_or_update_symbol` (I think we can just drop `_to_scope` without losing anything valuable.) I think this PR is the right place to do that rename, since adding the flags really adds "update" as a possibility for the first time.

Reply:
I'll do this, but probably not until the rest of the work is done and the review is looking like :+1:. Renaming it now might make rebases annoying.

Reply:
Doing this now ahead of rereview/merge.